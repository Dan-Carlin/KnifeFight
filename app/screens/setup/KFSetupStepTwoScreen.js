/*
KFSetupStepTwoScreen - View for the second setup screen.
*/

// External libraries
import React, { useState } from "react";
import { View } from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";

// Assets
import {
  BackGraphic,
  CloseGraphic,
  NextGraphic,
} from "../../assets/buttons/navigation";

// Components
import ColorButton from "../../components/ColorButton";
import OpacityButton from "../../components/OpacityButton";
import Screen from "../../components/Screen";
import Text from "../../components/Text";

// Resources
import gangColors from "../../data/gangColors";
import routes from "../../navigation/routes";
import strings from "../../config/strings";
import styles from "./KFSetupStepTwoStyles";

const background = require("../../assets/backgrounds/kf_background_xxxhdpi.png");

function KFSetupStepTwoScreen({ navigation }) {
  const [color, setColor] = useState(gangColors.NONE);
  const [colorIsSelected, setColorIsSelected] = useState(false);

  const setGangColor = (value) => {
    if (value != gangColors.NONE) {
      setColorIsSelected(true);
    } else {
      setColorIsSelected(false);
    }
    setColor(value);
  };

  const storeGangColor = async (value) => {
    try {
      const jsonValue = JSON.stringify(value);
      await AsyncStorage.setItem("@gangColor", jsonValue);
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <Screen style={styles.screenContainer} background={background}>
      <View style={styles.topContainer}>
        <View style={styles.closeButton}>
          <OpacityButton
            Graphic={CloseGraphic}
            onPress={() => navigation.popToTop()}
          />
        </View>
      </View>
      <View style={styles.bodyContainer}>
        <Text style={styles.title}>{strings.setup_step_two_title}</Text>
        <Text style={styles.text}>{strings.setup_step_two_text}</Text>
        <View style={styles.colorGrid}>
          <View style={styles.colorRow}>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.PURPLE}
                isSelected={color == gangColors.PURPLE}
                onSelect={() => setGangColor(gangColors.PURPLE)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.YELLOW}
                isSelected={color == gangColors.YELLOW}
                onSelect={() => setGangColor(gangColors.YELLOW)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.GREEN}
                isSelected={color == gangColors.GREEN}
                onSelect={() => setGangColor(gangColors.GREEN)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.RED}
                isSelected={color == gangColors.RED}
                onSelect={() => setGangColor(gangColors.RED)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.ORANGE}
                isSelected={color == gangColors.ORANGE}
                onSelect={() => setGangColor(gangColors.ORANGE)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.BLUE}
                isSelected={color == gangColors.BLUE}
                onSelect={() => setGangColor(gangColors.BLUE)}
              />
            </View>
          </View>
          <View style={styles.colorRow}>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.BLACK}
                isSelected={color == gangColors.BLACK}
                onSelect={() => setGangColor(gangColors.BLACK)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.WHITE}
                isSelected={color == gangColors.WHITE}
                onSelect={() => setGangColor(gangColors.WHITE)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.PINK}
                isSelected={color == gangColors.PINK}
                onSelect={() => setGangColor(gangColors.PINK)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.BROWN}
                isSelected={color == gangColors.BROWN}
                onSelect={() => setGangColor(gangColors.BROWN)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.CYAN}
                isSelected={color == gangColors.CYAN}
                onSelect={() => setGangColor(gangColors.CYAN)}
              />
            </View>
            <View style={styles.colorButton}>
              <ColorButton
                Color={gangColors.GOLD}
                isSelected={color == gangColors.GOLD}
                onSelect={() => setGangColor(gangColors.GOLD)}
              />
            </View>
          </View>
        </View>
        <View style={styles.navContainer}>
          <OpacityButton
            style={styles.navButton}
            Graphic={BackGraphic}
            onPress={() => navigation.goBack()}
            width="45%"
          />
          {colorIsSelected && (
            <OpacityButton
              style={styles.navButton}
              Graphic={NextGraphic}
              onPress={() => {
                storeGangColor(color);
                navigation.navigate(routes.SETUP_STEP_THREE);
              }}
              width="45%"
            />
          )}
          {!colorIsSelected && (
            <NextGraphic style={styles.buttonDisabled} width="45%" />
          )}
        </View>
      </View>
      <View style={styles.bottomContainer} />
    </Screen>
  );
}

export default KFSetupStepTwoScreen;
