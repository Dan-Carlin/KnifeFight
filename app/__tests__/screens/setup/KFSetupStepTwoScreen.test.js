import React from "react";
import renderer from "react-test-renderer";

import KFSetupStepTwoScreen from "../../../screens/setup/KFSetupStepTwoScreen";

describe("Setup second screen tests", () => {
  it("renders second Setup screen correctly", () => {
    const tree = renderer.create(<KFSetupStepTwoScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // sets color value correctly when Color button tapped

  // sets checked state of buttons not selected to false

  // activates Next button when color is selected

  // stores color value when Next button is tapped
});
