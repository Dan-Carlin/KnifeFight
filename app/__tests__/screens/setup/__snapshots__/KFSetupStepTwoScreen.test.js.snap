// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup second screen tests renders second Setup screen correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "paddingTop": 20,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 0.25,
        }
      }
    >
      <View
        style={
          Object {
            "height": 50,
            "marginStart": 35,
            "marginTop": 20,
            "width": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.5,
          "justifyContent": "center",
          "paddingHorizontal": 25,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "fontFamily": "default",
              "fontSize": 50,
              "textAlign": "center",
            },
          ]
        }
      >
        Next...
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        Let's choose a gang color.
      </Text>
      <View
        style={
          Object {
            "height": "55%",
            "marginVertical": 6,
            "width": "95%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "margin": 4,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.15,
                "justifyContent": "center",
              }
            }
          >
            <SvgMock
              style={
                Object {
                  "height": "90%",
                  "position": "absolute",
                  "width": "90%",
                }
              }
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 90,
            "justifyContent": "space-between",
            "width": "90%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "45%",
            }
          }
        >
          <SvgMock />
        </View>
        <SvgMock
          style={
            Object {
              "opacity": 0.5,
            }
          }
          width="45%"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.25,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
